azurerm/resource_arm_automation_dsc_configuration_test.go

```diff
@@ -0,0 +1,135 @@
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAutomationDscConfigurationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).automationDscConfigurationClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_automation_dsc_configuration" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		accName := rs.Primary.Attributes["automation_account_name"]
+
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Automation Dsc Configuration: '%s'", name)
+		}
+
+		resp, err := conn.Get(ctx, resourceGroup, accName, name)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		return fmt.Errorf("Automation Dsc Configuration still exists:\n%#v", resp)
+
+	}
+
+	return nil
+}
+
+func testCheckAzureRMAutomationDscConfigurationExists(name string) resource.TestCheckFunc {
+
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		accName := rs.Primary.Attributes["automation_account_name"]
+
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Automation Dsc Configuration: '%s'", name)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).automationDscConfigurationClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := conn.Get(ctx, resourceGroup, accName, name)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Automation Dsc Configuration '%s' (resource group: '%s') does not exist", name, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on automationDscConfigurationClient: %s\nName: %s, Account name: %s, Resource group: %s OBJECT: %+v", err, name, accName, resourceGroup, rs.Primary)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMAutomationDscConfiguration_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_automation_account" "test" {
+  name                = "acctest-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku {
+    name = "Basic"
+  }
+}
+
+resource "azurerm_automation_dsc_configuration" "test" {
+  name                    = "test"
```

Comment:
This is already the case for the automation account. Resources in scope of this pr cannot exist without the automation account so there is no need to change this

Reply:
Hi @bgelens, 

We name all resources in a acceptance test `acctest`. As there is no reason not to in this case it would be great if we could change it to align with our testing conventions.